storage/remote/queue_manager.go

```diff
@@ -1668,6 +1673,15 @@ func sendWriteRequestWithBackoff(ctx context.Context, cfg config.QueueConfig, l
 			level.Debug(l).Log("msg", "retry-after cannot be in past, retrying using default backoff mechanism")
 		}
 
+		// We should never reshard for a recoverable error; increasing shards could
+		// make the problem worse, particularly if we're getting rate limited.
+		//
+		// reshardDisableTimestamp holds the unix timestamp until which resharding
+		// is diableld. We'll update that timestamp if the period we were just told
+		// to sleep for is newer than the existing disabled timestamp.
+		reshardWaitPeriod := time.Now().Add(time.Duration(sleepDuration) * 2)
```

Comment:
Rather than blindly doubling the sleep duration, it's probably a better test to set this to sleepDuration plus the timeout for requests.

Reply:
do we want to set the `reshardDisableTimestamp` only for 429s?

Reply:
Although the PR description only mentions 429s, retries triggered by 500 errors are unlikely to be a great time to increase throughput either.

Reply:
Hmm, how about when a network link is slow, resulting in timeouts. More clients trying to send the same data might help with better throughput overall?

Reply:
That is just about believable, if you have the timeout set to something below 300ms (assuming terrestrial communication).
Don't set the timeout to 300ms.

Reply:
> Hmm, how about when a network link is slow, resulting in timeouts. More clients trying to send the same data might help with better throughput overall?

I'm not sure I understand how this might improve throughput; if the problem is a slow network link, wouldn't more concurrent requests cause more network pressure on the slow link and make things potentially worse?

Reply:
Which timeout are you referring to?

Reply:
> slow network link

I think it's clearer for a network link with large packet drops. In that case, more clients trying to send the same data might be better throughput.

Reply:
> Which timeout are you referring to?

`remote_timeout` in [remote_write](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write)